### PR TITLE
Break HTTP routes into multiple servers

### PR DESCRIPTION
**Pull request by joshklop**
For access control purposes.

### CODE REVIEW

**Comment by joshklop** [comment]
This is probably easiest to review all at once, rather than commit-by-commit. We can squash merge.

**Comment by omerfirmak** [comment]
We can put Http & ws rpc on the same port, WDYT?

**Comment by joshklop** [comment]
Rebased and made the adjustments mentioned in my last message. If you don't mind taking another quick look @omerfirmak, I would really appreciate it. 

Note that all services except synchronization are disabled by default now. Enabling them all with default ports would look something like:

```
juno --http --ws --metrics --pprof --grpc
```

**Review by omerfirmak — APPROVED** [review]

File: node/http.go [hunk 1]
```diff
@@ -16,94 +15,115 @@ import (
...
+		srv: &http.Server{
+			Addr:    listener.Addr().String(),
+			Handler: mux,
+			// ReadTimeout also sets ReadHeaderTimeout and IdleTimeout.
+			ReadTimeout: 30 * time.Second,
+		},
+		listener: listener,
+	}
+}
+
```
Comment (omerfirmak) [inline node/http.go:126]: I dont think we will ever use these in tests, right?
So can we create the listener in these functions so `node.New` has less copy-pasta?

Comment (omerfirmak) [reply node/http.go]: see https://github.com/NethermindEth/juno/pull/1087/commits/7fff1473398783135ccbe821edd6fc9a2b3c5904
Feel free to revert

Comment (joshklop) [reply node/http.go]: LGTM. Thanks

**Pull request merged**